source/elements/oneMKL/source/domains/spblas/data_types/create_coo_matrix.rst

```diff
@@ -0,0 +1,241 @@
+particular, the ``create_coo_matrix`` API must handle the case where the number
+of non-zero elements in the matrix are unknown and the matrix arrays are
+``nullptr`` or zero-sized ``sycl::buffer`` objects as applicable. This behavior
+is currently left to be implementation-defined, but may be clarified in the
+oneMKL Specification in the future.
+
+See :ref:`onemkl_sparse_matrix_handle`.
+
+.. _onemkl_sparse_create_coo_matrix_buffer:
+
+create_coo_matrix (Buffer version)
+----------------------------------
+
+.. rubric:: Syntax
+
+.. code-block:: cpp
+
+   namespace oneapi::mkl::sparse {
+
+       template <typename dataType, typename indexType>
+       void create_coo_matrix (sycl::queue                          &queue,
+                               oneapi::mkl::sparse::matrix_handle_t *p_smhandle,
+                               std::int64_t                         num_rows,
+                               std::int64_t                         num_cols,
+                               std::int64_t                         nnz,
+                               index_base                           index,
+                               sycl::buffer<indexType, 1>           &row_ind,
+                               sycl::buffer<indexType, 1>           &col_ind,
+                               sycl::buffer<dataType, 1>            &val);
+
+   }
+
+.. container:: section
+
+   .. rubric:: Template parameters
+
+   dataType
+      See :ref:`supported template types<onemkl_sparse_supported_types>`.
+
+   indexType
+      See :ref:`supported template types<onemkl_sparse_supported_types>`.
+
+.. container:: section
+
+   .. rubric:: Input parameters
+
+   queue
+      The SYCL command queue which will be used for SYCL kernels execution.
+
+   p_smhandle
+      The address of the ``p_smhandle`` object to be initialized. Must only be
+      called on an uninitialized ``matrix_handle_t`` object.
+
+   num_rows
+      Number of rows of the provided data ``val``. Must be at least 0.
+
+   num_cols
+      Number of columns of the provided data ``val``. Must be at least 0.
+
+   nnz
+      Number of non-zero entries in the matrix (which may include explicit
+      zeros). Must be at least 0.
+
+   index
+      Indicates how input arrays are indexed. The possible options are described
+      in :ref:`onemkl_enum_index_base` enum class.
+
+   row_ind
+      Buffer of length ``nnz`` containing the row indices in ``index``-based
+      numbering. Refer to :ref:`onemkl_sparse_coo` format for detailed
+      description of ``row_ind``.
+
+   col_ind
+      Buffer of length ``nnz`` containing the column indices in ``index``-based
+      numbering. Refer to :ref:`onemkl_sparse_coo` format for detailed
+      description of ``col_ind``.
+
+   val
+      Buffer of length ``nnz`` containing non-zero elements (and possibly
+      explicit zeros) of the input matrix. Refer to :ref:`onemkl_sparse_coo`
+      format for detailed description of ``val``.
```

Comment:
Here is where I think being explicit about zero-sized `sycl::buffer` or in case of USM, `nullptr`, is needed. I believe there are only 2 possible cases here:
1. `nnz = 0` and `row_ind`, `col_ind`, `val` are all zero-sized `sycl::buffers` / USM `nullptr`s, and the matrix handle may be considered as partially defined.
2. `nnz > 0`, in which case `row_ind`, `col_ind`, `val` must all be non-`nullptr` with correctly allocated sizes (at least length `nnz`) and the matrix handle is considered fully defined.

Reply:
Good point. I would have a preference for option 2 but option 1 is closer to cuSPARSE and rocSPARSE behavior from what I understand so will go with the first one for now.